IdentityCore/src/MSIDError.m

```diff
@@ -42,14 +42,19 @@
 NSString *MSIDHttpErrorCodeDomain = @"MSIDHttpErrorCodeDomain";
 NSString *MSIDInvalidTokenResultKey = @"MSIDInvalidTokenResultKey";
 
-NSError *MSIDCreateError(NSString *domain, NSInteger code, NSString *errorDescription, NSString *oauthError, NSString *subError, NSError *underlyingError, NSUUID *correlationId, NSDictionary *additionalUserInfo)
+NSError *MSIDCreateError(NSString *domain, NSInteger code, NSString *errorDescription, NSString *oauthError, NSString *subError, NSError *underlyingError, NSUUID *correlationId, NSDictionary *additionalUserInfo, BOOL logErrorDescription)
 {
     id<MSIDErrorConverting> errorConverter = MSIDErrorConverter.errorConverter;
 
     if (!errorConverter)
     {
         errorConverter = MSIDErrorConverter.defaultErrorConverter;
     }
+    
+    if (logErrorDescription)
+    {
+        MSID_LOG_WITH_CORR(MSIDLogLevelError, correlationId, @"%@", errorDescription);
```

Comment:
Will it be better to add something like the following to the logs, such that it explicitly tell where this line comes from? :)

`@"Creating Error with description: %@", errorDescription`

Reply:
Sure, addressed :)